Assets/Scripts/LevelGenerator.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+namespace DefaultNamespace
+{
+    public class LevelGenerator : MonoBehaviour
+    {
+        [SerializeField] private int _piecesCount = 5;
+        [SerializeField] private float _minPieceSize = 5f;
+        [SerializeField] private float _maxPieceSize = 25f;
+        [SerializeField] private GameObject _dir1Piece;
+        [SerializeField] private GameObject _dir2Piece;
+        [SerializeField] private Vector3 _piecesShift = new Vector3(-0.5f, 0, 0.5f);
+        [SerializeField] private GameObject _finishLine;
+        [SerializeField] private GameObject _coin;
+
+        private void Start()
+        {
+            bool dir1 = true;
+            Vector3 startPoint = new Vector3(0, -1, 0);
+            for (int i = 0; i < _piecesCount; i++)
+            {
+                float pieceSize = Random.Range(_minPieceSize, _maxPieceSize);
+                var coinDistance = Random.Range((-pieceSize / 2) + 0.5f, (pieceSize / 2) - 0.5f);
+                
+                if (dir1)
+                {
+                    var piece = Instantiate(_dir1Piece);
```

Comment:
code for `if` and `else` is effectively the same
you should think how you could extract common logic